### DISCUSSION TITLE
ESP32-2432S032 and ESP32-1732S019

### DISCUSSION THREAD

**Discussion by chromstyler** [BitMaker-hub/NerdMiner_v2#167] [Ideas]
i own several ESP32-2432S032 and ESP32-1732S019 boards
is there a chance to get nerdminer working on them too ?
as im right, there is only a modification in display driver needed to the actual ESP32-2432S028R 2,8" support or ?

**Comment by nitroxgas** [comment]
Would you mind to give us a link where to buy this boards, so we could check the compatibility ?

**Comment by chromstyler** [comment]
ESP32-1732S019
https://de.aliexpress.com/item/1005005734401975.html

ESP32-2432S032
https://de.aliexpress.com/item/1005005633096413.html

**Comment by sergang8888** [comment]
https://github.com/kpeeem/3.2inch_ESP32-2432S032

**Reply by chromstyler** [reply]
Thank you but i dont know what to Do need more advise

**Comment by ZuluDrag** [comment]
Have the same board ESP32-1732S019. As of one ebay lot https://www.ebay.co.uk/itm/325897862335 firmware should be the same as Lilygo S3 TDisplay. But after flashing it I have a black screen.
Any ideas on getting it working? :)

**Reply by pep987** [reply]
Did you find an answer in the end, I have the same board and same issue.

**Reply by ZuluDrag** [reply]
eBay seller doesn't want to share his f/w version :D Still waiting for someone to collect needed drivers and taking everything together. I had no time for that to do it myself :(

**Reply by AlVirtu** [reply]
[Webflasher-NerdMiner_v2](https://alvirtu.github.io/Webflasher-NerdMiner_v2-version-hardware-sha256/) for ESP32-1732S019N.

**Reply by Pazuzo** [reply]
> [Webflasher-NerdMiner_v2](https://alvirtu.github.io/Webflasher-NerdMiner_v2-version-hardware-sha256/) for ESP32-1732S019N.

Great work !

**Reply by Chiron69** [reply]
**reboot holding button under for 19 sec. Then flash correct ROM. Peace**

**Comment by pep987** [comment]
Did you ever get a resolution for the ESP32-1732S019? I have the same issue, I think its based on the S3 but needs custom drivers for the screen.

**Reply by ZuluDrag** [reply]
It's ESP32-WROOM-1 based, but different screen driver and different pins for buttons, as I know. But still, no opensourced Nerdminer version for it. eBay still has one seller that has already modified firmware...

**Reply by djcasper1975** [reply]
i bought one of his devices and tried to update it now its dead. he never said it was his own custom firmware now im stuck with a nice home made trangle box with a black screen.

**Reply by MyOwn2C** [reply]
You can always wipe the flash and start over

**Reply by djcasper1975** [reply]
Theres no support for the screen.. i did what some posts  said with the other one and that worked but if only there was the other device listed i could recompile and use the other one too.

**Reply by Chiron69** [reply]
> You can always wipe the flash and start over

Yup

**Comment by gwisp** [comment]
The library pins and actual pins differ, look up "CYD " as the pins are defined and there is a custom user_setup.h to make the screen work.
it uses both H and L Spi buses, one for the screen and the other for touch.

**Comment by Leonardoal26** [comment]
Has anyone found a solution?

**Reply by djcasper1975** [reply]
heres 1 fixed https://rapidshare.io/6ps/esp-2432S028_2USB.zip  not sure how to do the bigger screen 1 as i just followed instructions here on how to fix with 2 usb on

**Comment by Pazuzo** [comment]
I`m also looking for Nerdminer for ESP32-1732S019

**Reply by djcasper1975** [reply]
me too

**Comment by djcasper1975** [comment]
Ive noticed theres a flash file for ESP32-2432S032 screen is a bit funny before wifi is setup but it works yeay.  is there going to be a version for ESP32-1732S019? would make lots of people happy who dont want to pay lilygo prices

**Reply by Pazuzo** [reply]
how did you flash it? Itried all with no success with the screen :(

**Reply by djcasper1975** [reply]
I used esp tool online and added 1 extra file from the origional files i added 0xe000_boot_app0.bin

**Reply by djcasper1975** [reply]
heres the fixed version i just made. colours are great now.
https://rapidshare.io/6ps/esp-2432S028_2USB.zip

**Comment by djcasper1975** [comment]
I used esp tool online and added 1 extra file from the origional files i added 0xe000_boot_app0.bin

Hope this helps. I posted my files.



Sent from my Galaxy



-------- Original message --------
From: Pazuzo ***@***.***>
Date: 03/04/2024 17:32 (GMT+00:00)
To: BitMaker-hub/NerdMiner_v2 ***@***.***>
Cc: djcasper1975 ***@***.***>, Comment ***@***.***>
Subject: Re: [BitMaker-hub/NerdMiner_v2] ESP32-2432S032 and ESP32-1732S019 (Discussion #167)


how did you flash it? Itried all with no success with the screen :(

**Reply by Pazuzo** [reply]
i found platform info for ESP32-1732S019 , trying to build but i`m not good in this.

[env:cyd1732]
platform = espressif32
board = esp32-s3-devkitm-1
framework = arduino
lib_ldf_mode = deep+
monitor_filters = esp32_exception_decoder
monitor_speed = 115200
upload_speed = 460800
board_build.partitions = min_spiffs.csv
build_flags = 
    -DST7789_DRIVER
    -DTFT_WIDTH=170
    -DTFT_HEIGHT=320
    -DTFT_MISO=-1
    -DTFT_MOSI=13
    -DTFT_SCLK=12
    -DTFT_CS=10
    -DTFT_DC=11
    -DTFT_RST=1
    -DTFT_BL=14
    -DTFT_BACKLIGHT_ON=HIGH
    -DTFT_BACKLIGHT_OFF=LOW
    -DTOUCH_CS=-1
    -DLOAD_GLCD
    -DSPI_FREQUENCY=65000000
    -DSPI_READ_FREQUENCY=20000000
    -DSPI_TOUCH_FREQUENCY=2500000
    -DTFT_INVERSION_ON

**Reply by djcasper1975** [reply]
this is all new to me too hope you manage to sort it this little one is doing my head in. i will also see if i can understand how all this works to change it all. i dont know why ebay sellers think they can modify something and think because they can do it on a cheaper board before other is makes them more money but like i told the seller firstly he solder new usb on case as its home made and only use power 2 wires so i had to solder 2 more wires so program it. then when i did the screen didnt work after but its mining at 60-70 k with no screen lol. so nobody can ever update his device and if they set it up wrong they cant do anything with it at all because they cant reflash them. its in his benifit to share this so the store can use it and people can update his device which is unique anyway if he added 2 wires and shared the screen fix info and files.... anyhow hopefully you can do this......

**Reply by Pazuzo** [reply]
Tomorrow I plan to delve into these things,  this is a new area for me also.

**Reply by djcasper1975** [reply]
i added that to the bottom of the platformio.ini and it was there to build with but had tons of errors after i tried.

**Reply by Pazuzo** [reply]
Do you have any update. or Is there any progress?

**Comment by okeeffm** [comment]
Hi, I’m the eBay seller that modified the code to get these cheaper boards to work.  A eBay buyer made me aware there was a discussion thread - my apologies, I’m fairly new to GitHub and the first time I used it was to download the Nerd Miner Code!

I don’t have time this week to list out all the changes I made but the main issue with these 1.9” ESP32 S3 boards was getting TFT_eSPI library config working with this board.  I came across this issue raised against the Bodmer TFT_eSPI - https://github.com/Bodmer/TFT_eSPI/discussions/2939

The proper solution is getting the setup.h and user_setup.h (and platformio.ini) files set up properly.  However, I couldn’t  get this to work so I worked out a work-around. This work-around involves several  changes to the project - I’d rather give you a clean solution so I’ll come back to this thread next week sometime.

**Reply by Leonardoal26** [reply]
Hello, can you provide us with the bin files for this purpose please?

**Reply by rgegusch** [reply]
will you come back soon?

**Comment by Pazuzo** [comment]
> Hello, can you provide us with the bin files for this purpose please?

I think he won't share

**Reply by djcasper1975** [reply]
[esp-2432S028_2USB.zip](https://github.com/BitMaker-hub/NerdMiner_v2/files/15081934/esp-2432S028_2USB.zip)

**Reply by PeterLin74** [reply]
> [esp-2432S028_2USB.zip](https://github.com/BitMaker-hub/NerdMiner_v2/files/15081934/esp-2432S028_2USB.zip)

Hi Casper,
I've try the firmware you provided,
but still cant't work.
Is there any update for ESP32-1732S019?

**Comment by djcasper1975** [comment]
i have attached files lots of times here it is again

________________________________

**Reply by Pazuzo** [reply]
2432S028_2USB is in another post, we talking about 1732S019 :)

**Comment by Henbal-Invest** [comment]
Anyone has some News on ESP32-1732S019?

**Reply by sardor1122** [reply]
https://github.com/Bodmer/TFT_eSPI/discussions/2939#discussioncomment-10704255

**Comment by nebula-21** [comment]


**Reply by Pazuzo** [reply]
It's just a copy , no ESP32-1732S019 inside :)

**Comment by iTrooper5118** [comment]
Has there been any progress in getting ESP32-2432S032 working yet? I can see NerdMinerAP showing up, but the screen itself is pretty dead. I would have thought it would have just been a matter of changing some configuration somewhere to get the screen going, but so far no solution :(

**Comment by m8ax** [comment]
probe this firmware https://m8ax.github.io/MvIiIaX-NerdMiner_V2-DeV/

**Reply by AlVirtu** [reply]
There is no support for ESP32-2432S032 and ESP32-1732S019 boards here.
Here you can flash ESP32-1732S019 board - [Webflasher-NerdMiner_v2](https://alvirtu.github.io/Webflasher-NerdMiner_v2-version-hardware-sha256/) for ESP32-1732S019N.
